apax/train/checkpoints.py

```diff
@@ -14,13 +14,56 @@
 log = logging.getLogger(__name__)
 
 
-def load_state(model, params, tx, ckpt_dir):
+def check_for_ensemble(params):
+    flat_params = flatten_dict(params)
+    shapes = [v.shape[0] for v in flat_params.values()]
+    is_ensemble = shapes == shapes[::-1]
```

Comment:
Why is it an ensemble if the shape is equal to its reversed?

Reply:
I suppose a clearer way of checking would be to see if all the first dimensions are the same